mtop-client/src/dns/client.rs

```diff
@@ -177,9 +178,14 @@ where
     U: ClientFactory<SocketAddr, UdpConnection> + Send + Sync + 'static,
     T: ClientFactory<SocketAddr, TcpConnection> + Send + Sync + 'static,
 {
-    async fn resolve(&self, name: Name, rtype: RecordType, rclass: RecordClass) -> Result<Message, MtopError> {
+    async fn resolve(
+        &self,
+        id: MessageId,
```

Comment:
The `id` parameter is now part of the interface because we need to know the ID associated with the request to be able to provide mocked responses when testing `DefaultDnsClient`.